Controller certificate watch fails in secondary datacenter

<!--- 

IMPORTANT: If you are having problems installing the Consul API Gateway, before filing an issue, please ensure you have followed each step of the documented installation procedure detailed at https://www.consul.io/docs/api-gateway/api-gateway-usage#installation.

When filing a bug, please include the following headings if possible. Any example text in this template can be deleted.

--->

### Overview of the Issue

<!--- Please describe the issue you are having and how you encountered the problem. --->

This issue is a contributing factor to https://github.com/hashicorp/consul-api-gateway/issues/300.

The controller fails to initialize a watch on root certificates when starting up in a secondary datacenter. This is because the primary datacenter needs to be included when creating the watch but is not today.

https://github.com/hashicorp/consul-api-gateway/blob/c35994bcb8fbcec332040acd1b7c6bb1d72e9d3e/internal/consul/certmanager.go#L215-L217

### Reproduction Steps

<!--- 

In order to effectively and quickly resolve the issue, please provide exact steps that allow us the reproduce the problem. If no steps are provided, then it will likely take longer to get the issue resolved. An example that you can follow is provided below. 

Steps to reproduce this issue, eg:

1. When creating a gateway with the following configuration:
```
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
...
```
1. View error

--->

1. Create a federated setup by following the guide [Federation Between Kubernetes Clusters](https://www.consul.io/docs/k8s/deployment-configurations/multi-cluster/kubernetes). When installing, ensure API Gateway is enabled in both the primary and secondary clusters (`apiGateway.enabled: true` in values.yaml).
2. View logs of the API gateway controller in the secondary datacenter (`kubectl logs ...`)

### Logs

<details>
  <summary>Logs</summary>

```log
2022-09-06T19:58:26.971Z [INFO]  k8s/logger.go:30: consul-api-gateway-server.controller-runtime: Starting workers: controller=gateway controllerGroup=gateway.networking.k8s.io controllerKind=Gateway info="Starting workers" worker count=1
2022-09-06T19:58:26.971Z [INFO]  k8s/logger.go:30: consul-api-gateway-server.controller-runtime: Starting workers: controller=tcproute controllerGroup=gateway.networking.k8s.io controllerKind=TCPRoute info="Starting workers" worker count=1
2022-09-06T20:08:30.888Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:18:51.442Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:39:13.030Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:49:28.020Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T20:59:41.606Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T21:40:44.984Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: i/o deadline reached)" retry=5s
2022-09-06T21:51:03.780Z [ERROR] watch/plan.go:95: consul-api-gateway-server.cert-manager.watch: Watch errored: type=connect_roots error="Unexpected response code: 500 (rpc error making call: rpc error making call: i/o deadline reached)" retry=5s
```

</details>

### Expected behavior

<!--- What was the expected result after following the reproduction steps? --->

Controller successfully starts up (including cert watch initiated as part of startup)

### Environment details

If not already included, please provide the following:
- consul-api-gateway version: v0.4.0
- configuration used to deploy the gateway controller:
```yaml
apiGateway:
  enabled: true
  image: "hashicorp/consul-api-gateway:0.4.0"
```

Additionally, please provide details regarding the Kubernetes Infrastructure, as shown below:
- Consul Server version: v1.13.1
- Consul-K8s version: 0.47.1
- Cloud Provider (If self-hosted, the Kubernetes provider utilized): GKE

### Additional Context

<!---
Additional context on the problem. Docs, links to blogs, or other material that lead you to discover this issue or were helpful in troubleshooting the issue. 
--->

--- a/.changelog/368.txt
+++ b/.changelog/368.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+Fix failing root certificate watch for controller when deployed in secondary federated datacenter.
+```

--- a/internal/commands/exec/command.go
+++ b/internal/commands/exec/command.go
@@ -14,9 +14,10 @@ import (
 	"github.com/google/uuid"
 	"github.com/mitchellh/cli"
 
-	"github.com/hashicorp/consul-api-gateway/internal/common"
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul-api-gateway/internal/common"
 )
 
 // https://github.com/hashicorp/consul-k8s/blob/24be51c58461e71365ca39f113dae0379f7a1b7c/control-plane/connect-inject/container_init.go#L272-L306
@@ -39,10 +40,11 @@ type Command struct {
 	isTest bool
 
 	// Consul params
-	flagConsulHTTPAddress string // Address for Consul HTTP API.
-	flagConsulHTTPPort    int    // Port for Consul HTTP communication
-	flagConsulCACertFile  string // Root CA file for Consul
-	flagConsulXDSPort     int    // Port for Consul xDS communication
+	flagConsulHTTPAddress       string // Address for Consul HTTP API.
+	flagConsulHTTPPort          int    // Port for Consul HTTP communication
+	flagConsulCACertFile        string // Root CA file for Consul
+	flagConsulXDSPort           int    // Port for Consul xDS communication
+	flagConsulPrimaryDatacenter string // Name of the primary datacenter if deploying into a secondary datacenter
 
 	// Gateway params
 	flagGatewayID        string // Gateway iD.
@@ -86,6 +88,7 @@ func (c *Command) init() {
 		c.flagSet.IntVar(&c.flagConsulHTTPPort, "consul-http-port", 8500, "Port of Consul HTTP server.")
 		c.flagSet.IntVar(&c.flagConsulXDSPort, "consul-xds-port", 8502, "Port of Consul xDS server.")
 		c.flagSet.StringVar(&c.flagConsulCACertFile, "consul-ca-cert-file", "", "CA Root file for Consul.")
+		c.flagSet.StringVar(&c.flagConsulPrimaryDatacenter, "consul-primary-datacenter", "", "Primary datacenter if deploying into a secondary datacenter.")
 	}
 	{
 		// Envoy
@@ -163,11 +166,12 @@ func (c *Command) Run(args []string) (ret int) {
 	}
 
 	return RunExec(ExecConfig{
-		Context:      c.ctx,
-		Logger:       logger,
-		LogLevel:     c.flagLogLevel,
-		ConsulClient: consulClient,
-		ConsulConfig: *cfg,
+		Context:           c.ctx,
+		Logger:            logger,
+		LogLevel:          c.flagLogLevel,
+		ConsulClient:      consulClient,
+		ConsulConfig:      *cfg,
+		PrimaryDatacenter: c.flagConsulPrimaryDatacenter,
 		AuthConfig: AuthConfig{
 			Method:    c.flagACLAuthMethod,
 			Namespace: c.flagAuthMethodNamespace,

--- a/internal/commands/exec/exec.go
+++ b/internal/commands/exec/exec.go
@@ -11,10 +11,11 @@ import (
 
 	"golang.org/x/sync/errgroup"
 
-	"github.com/hashicorp/consul-api-gateway/internal/consul"
-	"github.com/hashicorp/consul-api-gateway/internal/envoy"
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul-api-gateway/internal/consul"
+	"github.com/hashicorp/consul-api-gateway/internal/envoy"
 )
 
 type AuthConfig struct {
@@ -43,14 +44,15 @@ type EnvoyConfig struct {
 }
 
 type ExecConfig struct {
-	Context       context.Context
-	Logger        hclog.Logger
-	LogLevel      string
-	ConsulClient  *api.Client
-	ConsulConfig  api.Config
-	AuthConfig    AuthConfig
-	GatewayConfig GatewayConfig
-	EnvoyConfig   EnvoyConfig
+	Context           context.Context
+	Logger            hclog.Logger
+	LogLevel          string
+	ConsulClient      *api.Client
+	ConsulConfig      api.Config
+	AuthConfig        AuthConfig
+	GatewayConfig     GatewayConfig
+	EnvoyConfig       EnvoyConfig
+	PrimaryDatacenter string
 
 	// for testing only
 	isTest bool
@@ -140,6 +142,7 @@ func RunExec(config ExecConfig) (ret int) {
 		},
 	)
 	options := consul.DefaultCertManagerOptions()
+	options.PrimaryDatacenter = config.PrimaryDatacenter
 	options.SDSAddress = config.EnvoyConfig.SDSAddress
 	options.SDSPort = config.EnvoyConfig.SDSPort
 	options.Directory = "/certs"

--- a/internal/commands/server/command.go
+++ b/internal/commands/server/command.go
@@ -39,6 +39,8 @@ type Command struct {
 
 	flagConsulAddress string // Consul server address
 
+	flagPrimaryDatacenter string // Primary datacenter, may or may not be the datacenter this controller is running in
+
 	flagSDSServerHost string // SDS server host
 	flagSDSServerPort int    // SDS server port
 	flagMetricsPort   int    // Port for prometheus metrics
@@ -70,6 +72,7 @@ func (c *Command) init() {
 	c.flagSet.StringVar(&c.flagCASecret, "ca-secret", "", "CA Secret for Consul server.")
 	c.flagSet.StringVar(&c.flagCASecretNamespace, "ca-secret-namespace", "default", "CA Secret namespace for Consul server.")
 	c.flagSet.StringVar(&c.flagConsulAddress, "consul-address", "", "Consul Address.")
+	c.flagSet.StringVar(&c.flagPrimaryDatacenter, "primary-datacenter", "", "Name of the primary Consul datacenter")
 	c.flagSet.StringVar(&c.flagSDSServerHost, "sds-server-host", defaultSDSServerHost, "SDS Server Host.")
 	c.flagSet.StringVar(&c.flagK8sContext, "k8s-context", "", "Kubernetes context to use.")
 	c.flagSet.StringVar(&c.flagK8sNamespace, "k8s-namespace", "", "Kubernetes namespace to use.")
@@ -164,13 +167,14 @@ func (c *Command) Run(args []string) int {
 	}
 
 	return RunServer(ServerConfig{
-		Context:       context.Background(),
-		Logger:        logger,
-		ConsulConfig:  consulCfg,
-		K8sConfig:     cfg,
-		ProfilingPort: c.flagPprofPort,
-		MetricsPort:   c.flagMetricsPort,
-		isTest:        c.isTest,
+		Context:           context.Background(),
+		Logger:            logger,
+		ConsulConfig:      consulCfg,
+		K8sConfig:         cfg,
+		ProfilingPort:     c.flagPprofPort,
+		MetricsPort:       c.flagMetricsPort,
+		PrimaryDatacenter: c.flagPrimaryDatacenter,
+		isTest:            c.isTest,
 	})
 }
 

--- a/internal/commands/server/command.go
+++ b/internal/commands/server/command.go
@@ -122,6 +122,7 @@ func (c *Command) Run(args []string) int {
 	cfg.SDSServerHost = c.flagSDSServerHost
 	cfg.SDSServerPort = c.flagSDSServerPort
 	cfg.Namespace = c.flagK8sNamespace
+	cfg.PrimaryDatacenter = c.flagPrimaryDatacenter
 
 	consulCfg := api.DefaultConfig()
 	if c.flagCAFile != "" {

--- a/internal/commands/server/server.go
+++ b/internal/commands/server/server.go
@@ -24,12 +24,13 @@ import (
 )
 
 type ServerConfig struct {
-	Context       context.Context
-	Logger        hclog.Logger
-	ConsulConfig  *api.Config
-	K8sConfig     *k8s.Config
-	ProfilingPort int
-	MetricsPort   int
+	Context           context.Context
+	Logger            hclog.Logger
+	ConsulConfig      *api.Config
+	K8sConfig         *k8s.Config
+	ProfilingPort     int
+	MetricsPort       int
+	PrimaryDatacenter string
 
 	// for testing only
 	isTest bool
@@ -72,6 +73,8 @@ func RunServer(config ServerConfig) int {
 	controller.SetStore(store)
 
 	options := consul.DefaultCertManagerOptions()
+	options.PrimaryDatacenter = config.PrimaryDatacenter
+
 	certManager := consul.NewCertManager(
 		config.Logger.Named("cert-manager"),
 		consulClient,

--- a/internal/consul/certmanager.go
+++ b/internal/consul/certmanager.go
@@ -73,9 +73,10 @@ func init() {
 
 // CertManagerOptions contains the optional configuration used to initialize a CertManager.
 type CertManagerOptions struct {
-	Directory  string
-	SDSAddress string
-	SDSPort    int
+	Directory         string
+	PrimaryDatacenter string
+	SDSAddress        string
+	SDSPort           int
 }
 
 // DefaultCertManagerOptions returns the default options for a CertManager instance.
@@ -97,11 +98,12 @@ type CertManager struct {
 	consul *api.Client
 	logger hclog.Logger
 
-	service         string
-	directory       string
-	configDirectory string // only used for testing
-	sdsAddress      string
-	sdsPort         int
+	service           string
+	directory         string
+	configDirectory   string // only used for testing
+	primaryDatacenter string
+	sdsAddress        string
+	sdsPort           int
 
 	lock sync.RWMutex
 
@@ -130,14 +132,15 @@ func NewCertManager(logger hclog.Logger, consul *api.Client, service string, opt
 		options = DefaultCertManagerOptions()
 	}
 	manager := &CertManager{
-		consul:           consul,
-		logger:           logger,
-		sdsAddress:       options.SDSAddress,
-		sdsPort:          options.SDSPort,
-		service:          service,
-		configDirectory:  options.Directory,
-		directory:        options.Directory,
-		initializeSignal: make(chan struct{}),
+		consul:            consul,
+		logger:            logger,
+		primaryDatacenter: options.PrimaryDatacenter,
+		sdsAddress:        options.SDSAddress,
+		sdsPort:           options.SDSPort,
+		service:           service,
+		configDirectory:   options.Directory,
+		directory:         options.Directory,
+		initializeSignal:  make(chan struct{}),
 	}
 	manager.writeCerts = manager.persist
 	return manager
@@ -213,7 +216,8 @@ func (c *CertManager) Manage(ctx context.Context) error {
 	c.logger.Trace("running cert manager")
 
 	rootWatch, err := watch.Parse(map[string]interface{}{
-		"type": "connect_roots",
+		"datacenter": c.primaryDatacenter,
+		"type":       "connect_roots",
 	})
 	if err != nil {
 		return err

--- a/internal/k8s/builder/gateway.go
+++ b/internal/k8s/builder/gateway.go
@@ -80,12 +80,13 @@ func filterAnnotations(annotations map[string]string, allowed []string) map[stri
 }
 
 type GatewayDeploymentBuilder struct {
-	gateway                *gwv1beta1.Gateway
-	gwConfig               *v1alpha1.GatewayClassConfig
-	sdsHost                string
-	sdsPort                int
-	consulCAData           string
-	consulGatewayNamespace string
+	gateway                 *gwv1beta1.Gateway
+	gwConfig                *v1alpha1.GatewayClassConfig
+	sdsHost                 string
+	sdsPort                 int
+	consulCAData            string
+	consulGatewayNamespace  string
+	consulPrimaryDatacenter string
 }
 
 func NewGatewayDeployment(gw *gwv1beta1.Gateway) *GatewayDeploymentBuilder {
@@ -107,6 +108,10 @@ func (b *GatewayDeploymentBuilder) WithConsulCA(caData string) {
 
 func (b *GatewayDeploymentBuilder) WithConsulGatewayNamespace(namespace string) {
 	b.consulGatewayNamespace = namespace
+}
+
+func (b *GatewayDeploymentBuilder) WithPrimaryConsulDatacenter(datacenter string) {
+	b.consulPrimaryDatacenter = datacenter
 }
 
 func (b *GatewayDeploymentBuilder) Validate() error {
@@ -262,16 +267,17 @@ func (b *GatewayDeploymentBuilder) podSpec() corev1.PodSpec {
 func (b *GatewayDeploymentBuilder) execCommand() []string {
 	// Render the command
 	data := gwContainerCommandData{
-		ACLAuthMethod:    b.gwConfig.Spec.ConsulSpec.AuthSpec.Method,
-		ConsulHTTPAddr:   orDefault(b.gwConfig.Spec.ConsulSpec.Address, defaultConsulAddress),
-		ConsulHTTPPort:   orDefaultIntString(b.gwConfig.Spec.ConsulSpec.PortSpec.HTTP, defaultConsulHTTPPort),
-		ConsulGRPCPort:   orDefaultIntString(b.gwConfig.Spec.ConsulSpec.PortSpec.GRPC, defaultConsulXDSPort),
-		LogLevel:         orDefault(b.gwConfig.Spec.LogLevel, defaultLogLevel),
-		GatewayHost:      "$(IP)",
-		GatewayName:      b.gateway.Name,
-		GatewayNamespace: b.consulGatewayNamespace,
-		SDSHost:          b.sdsHost,
-		SDSPort:          b.sdsPort,
+		ACLAuthMethod:     b.gwConfig.Spec.ConsulSpec.AuthSpec.Method,
+		ConsulHTTPAddr:    orDefault(b.gwConfig.Spec.ConsulSpec.Address, defaultConsulAddress),
+		ConsulHTTPPort:    orDefaultIntString(b.gwConfig.Spec.ConsulSpec.PortSpec.HTTP, defaultConsulHTTPPort),
+		ConsulGRPCPort:    orDefaultIntString(b.gwConfig.Spec.ConsulSpec.PortSpec.GRPC, defaultConsulXDSPort),
+		LogLevel:          orDefault(b.gwConfig.Spec.LogLevel, defaultLogLevel),
+		GatewayHost:       "$(IP)",
+		GatewayName:       b.gateway.Name,
+		GatewayNamespace:  b.consulGatewayNamespace,
+		PrimaryDatacenter: b.consulPrimaryDatacenter,
+		SDSHost:           b.sdsHost,
+		SDSPort:           b.sdsPort,
 	}
 	if b.requiresCA() {
 		data.ConsulCAFile = consulCALocalFile
@@ -349,18 +355,19 @@ func (b *GatewayDeploymentBuilder) requiresCA() bool {
 }
 
 type gwContainerCommandData struct {
-	ConsulCAFile     string
-	ConsulCAData     string
-	ConsulHTTPAddr   string
-	ConsulHTTPPort   string
-	ConsulGRPCPort   string
-	ACLAuthMethod    string
-	LogLevel         string
-	GatewayHost      string
-	GatewayName      string
-	GatewayNamespace string
-	SDSHost          string
-	SDSPort          int
+	ConsulCAFile      string
+	ConsulCAData      string
+	ConsulHTTPAddr    string
+	ConsulHTTPPort    string
+	ConsulGRPCPort    string
+	ACLAuthMethod     string
+	LogLevel          string
+	GatewayHost       string
+	GatewayName       string
+	GatewayNamespace  string
+	PrimaryDatacenter string
+	SDSHost           string
+	SDSPort           int
 }
 
 // gwContainerCommandTpl is the template for the command executed by
@@ -385,6 +392,9 @@ exec /bootstrap/consul-api-gateway exec -log-json \
   -consul-xds-port  {{ .ConsulGRPCPort }} \
 {{- if .ACLAuthMethod }}
   -acl-auth-method {{ .ACLAuthMethod }} \
+{{- end }}
+{{- if .PrimaryDatacenter }}
+  -primary-datacenter {{ .PrimaryDatacenter }}
 {{- end }}
   -envoy-bootstrap-path /bootstrap/envoy.json \
   -envoy-sds-address {{ .SDSHost }} \

--- a/internal/k8s/builder/gateway.go
+++ b/internal/k8s/builder/gateway.go
@@ -394,7 +394,7 @@ exec /bootstrap/consul-api-gateway exec -log-json \
   -acl-auth-method {{ .ACLAuthMethod }} \
 {{- end }}
 {{- if .PrimaryDatacenter }}
-  -primary-datacenter {{ .PrimaryDatacenter }}
+  -consul-primary-datacenter {{ .PrimaryDatacenter }} \
 {{- end }}
   -envoy-bootstrap-path /bootstrap/envoy.json \
   -envoy-sds-address {{ .SDSHost }} \

--- a/internal/k8s/controller.go
+++ b/internal/k8s/controller.go
@@ -69,6 +69,7 @@ type Kubernetes struct {
 
 type Config struct {
 	CACert              string
+	PrimaryDatacenter   string
 	SDSServerHost       string
 	SDSServerPort       int
 	MetricsBindAddr     string
@@ -154,6 +155,7 @@ func (k *Kubernetes) Start(ctx context.Context) error {
 		Client:                gwClient,
 		Consul:                k.consul,
 		ConsulCA:              k.config.CACert,
+		PrimaryDatacenter:     k.config.PrimaryDatacenter,
 		SDSHost:               k.config.SDSServerHost,
 		SDSPort:               k.config.SDSServerPort,
 		Logger:                k.logger.Named("Reconciler"),

--- a/internal/k8s/reconciler/deployer.go
+++ b/internal/k8s/reconciler/deployer.go
@@ -18,29 +18,32 @@ import (
 
 // GatewayDeployer creates gateway deployments and services and ensures that they exist
 type GatewayDeployer struct {
-	client   gatewayclient.Client
-	consulCA string
-	sdsHost  string
-	sdsPort  int
+	client            gatewayclient.Client
+	consulCA          string
+	primaryDatacenter string
+	sdsHost           string
+	sdsPort           int
 
 	logger hclog.Logger
 }
 
 type DeployerConfig struct {
-	ConsulCA string
-	SDSHost  string
-	SDSPort  int
-	Logger   hclog.Logger
-	Client   gatewayclient.Client
+	ConsulCA          string
+	PrimaryDatacenter string
+	SDSHost           string
+	SDSPort           int
+	Logger            hclog.Logger
+	Client            gatewayclient.Client
 }
 
 func NewDeployer(config DeployerConfig) *GatewayDeployer {
 	return &GatewayDeployer{
-		client:   config.Client,
-		consulCA: config.ConsulCA,
-		sdsHost:  config.SDSHost,
-		sdsPort:  config.SDSPort,
-		logger:   config.Logger,
+		client:            config.Client,
+		consulCA:          config.ConsulCA,
+		primaryDatacenter: config.PrimaryDatacenter,
+		sdsHost:           config.SDSHost,
+		sdsPort:           config.SDSPort,
+		logger:            config.Logger,
 	}
 }
 
@@ -129,6 +132,7 @@ func (d *GatewayDeployer) Deployment(namespace string, config apigwv1alpha1.Gate
 	deploymentBuilder.WithClassConfig(config)
 	deploymentBuilder.WithConsulCA(d.consulCA)
 	deploymentBuilder.WithConsulGatewayNamespace(namespace)
+	deploymentBuilder.WithPrimaryConsulDatacenter(d.primaryDatacenter)
 	return deploymentBuilder.Build(currentReplicas)
 }
 

--- a/internal/k8s/reconciler/manager.go
+++ b/internal/k8s/reconciler/manager.go
@@ -68,6 +68,7 @@ type ManagerConfig struct {
 	Client                gatewayclient.Client
 	Consul                *api.Client
 	ConsulCA              string
+	PrimaryDatacenter     string
 	SDSHost               string
 	SDSPort               int
 	Store                 store.Store
@@ -78,11 +79,12 @@ type ManagerConfig struct {
 func NewReconcileManager(config ManagerConfig) *GatewayReconcileManager {
 	resolver := service.NewBackendResolver(config.Logger, config.ConsulNamespaceMapper, config.Client, config.Consul)
 	deployer := NewDeployer(DeployerConfig{
-		ConsulCA: config.ConsulCA,
-		SDSHost:  config.SDSHost,
-		SDSPort:  config.SDSPort,
-		Logger:   config.Logger,
-		Client:   config.Client,
+		ConsulCA:          config.ConsulCA,
+		PrimaryDatacenter: config.PrimaryDatacenter,
+		SDSHost:           config.SDSHost,
+		SDSPort:           config.SDSPort,
+		Logger:            config.Logger,
+		Client:            config.Client,
 	})
 
 	return &GatewayReconcileManager{

